File: 3 - Linguagem C++/3 - Vetor - ler-mercadorias-preco-compra-venda-calcular-lucros-e-total.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main(){
    int N, lucroAbaixo = 0, lucroMed = 0, lucroAcima = 0;
    double totalCompra = 0, totalVenda = 0, lucroTotal = 0;

    cout << "Serao digitados dados de quantos produtos? ";
    cin >> N;

    string nomes[N];
    double precoCompra[N], precoVenda[N], lucro[N];

    for (int i=0; i<N; i++) {
        cout << "Produto " << i + 1 << endl;
        cout << "Nome: ";
        cin >> nomes[i];
        cout << "Preco de compra: ";
        cin >> precoCompra[i];
        cout << "Preco de venda: ";
        cin >> precoVenda[i];
    }

    for (int i=0; i<N; i++){
      lucro[i] = (precoVenda[i] - precoCompra[i]) / precoCompra[i] * 100.0; //lucro em relacao ao preco de compra
    }

    for (int i=0; i<N; i++){
        if(lucro[i] < 10){
            lucroAbaixo++;
        }
        else if(lucro[i] < 20) {
            lucroMed++;
        }
        else{
            lucroAcima++;
        }
    }

    for (int i=0; i<N; i++){
        totalCompra = totalCompra + precoCompra[i];
        totalVenda = totalVenda + precoVenda[i];
    }

    lucroTotal = totalVenda - totalCompra;


    cout << endl << "RELATORIO: " << endl;
    cout << "Lucro abaixo de 10%: " << lucroAbaixo << endl;
    cout << "Lucro entre 10% e 20%: " << lucroMed << endl;
    cout << "Lucro acima de 20%: " << lucroAcima  << endl;
    cout << fixed << setprecision(2);
    cout << "Valor total de compra: " << totalCompra  << endl;
    cout << "Valor total de venda: " << totalVenda  << endl;
    cout << "Lucro total: " << lucroTotal  << endl;

    return 0;
}

File: 4 - JAVA/3 - Vetor - ler-mercadorias-preco-compra-venda-calcular-lucros-e-total.java
import java.util.Locale;
import java.util.Scanner;

public class Main {
	
	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int N, i, lucroAbaixo=0, lucroMedio=0, lucroAcima=0;
		double valorTotCompra=0, valorTotVenda=0, lucroTotal;
		
		System.out.print("Serao digitados dados de quantos produtos? ");
		N = sc.nextInt(); 
		
		double[] compras = new double[N];
		double[] vendas = new double[N];
		String[] nomes = new String[N];
		double[] lucros = new double[N];
		
		for (i=0; i<N; i++) {
			System.out.printf("Produto %d:\n ", i + 1);
			System.out.print("Nome: ");
			sc.nextLine();
        	nomes[i] = sc.nextLine();
			System.out.print("Preco de compra: ");
			compras[i] = sc.nextDouble();
			System.out.print("Preco de venda: ");
			vendas[i] = sc.nextDouble();
		}
		
		for (i=0; i<N; i++) {
			lucros[i] = (vendas[i] - compras[i]) / compras[i] * 100;
		}
		
		for (i=0; i<N; i++) {
			if (lucros[i] < 10.0) {
				lucroAbaixo++;
			}
			else if (lucros[i] < 20.0) {
				lucroMedio++;
			}
			else {
				lucroAcima++;
			}
		}
		
		for (i=0; i<N; i++) {
			valorTotCompra = valorTotCompra + compras[i];
			valorTotVenda = valorTotVenda + vendas[i];
		}
		
		lucroTotal = valorTotVenda - valorTotCompra;
		
		System.out.println("\nRELATORIO:");
		System.out.printf("Lucro abaixo de 10%%: %d\n", lucroAbaixo);
	    System.out.printf("Lucro entre 10%% e 20%%: %d\n", lucroMedio);
	    System.out.printf("Lucro acima de 20%%: %d\n", lucroAcima);
	    System.out.printf("Valor total de compra: %.2f\n", valorTotCompra);
	    System.out.printf("Valor total de venda: %.2f\n", valorTotVenda);
	    System.out.printf("Lucro total: %.2f\n", lucroTotal);

		
		sc.close();
	}
}